providers/implementations/ciphers/ciphercommon.c

```diff
@@ -435,10 +436,8 @@ int ossl_cipher_generic_block_final(void *vctx, unsigned char *out,
         return 0;
     }
 
-    if (ctx->pad && !ossl_cipher_unpadblock(ctx->buf, &ctx->bufsz, blksz)) {
-        /* ERR_raise already called */
-        return 0;
-    }
+    /* ERR_raise not called */
+    gp &= ctx->pad & ossl_cipher_unpadblock(ctx->buf, &ctx->bufsz, blksz);
```

Comment:
Changing this from && to & is probably not going to do what you want, the compiler is probably going to create the same code.

Reply:
Fixed in commit `526350c`.